mongock-spring/src/main/java/com/github/cloudyrock/mongock/SpringBootMongock.java

```diff
@@ -0,0 +1,57 @@
+package com.github.cloudyrock.mongock;
+
+import com.mongodb.MongoClient;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.BeansException;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.context.ApplicationContext;
+
+public class SpringBootMongock extends Mongock implements ApplicationRunner {
+
+    ApplicationContext springContext;
+
+    protected SpringBootMongock(ChangeEntryRepository changeEntryRepository,
+        MongoClient mongoClient,
+        ChangeService changeService,
+        LockChecker lockChecker) {
+      super(changeEntryRepository, mongoClient, changeService, lockChecker);
+    }
+
+    /**
+     * @see ApplicationRunner#run(ApplicationArguments)
+     * @see Mongock#execute()
+     */
+    @Override
+    public void run(ApplicationArguments args) throws Exception {
+      execute();
+    }
+
+    /**
+     * Overrides the internal argument resolution strategy to allow spring to inject the appropriate parameters.
+     *
+     * @see Mongock#executeChangeSetMethod(Method, Object)
+     */
+    @Override
+    protected void executeChangeSetMethod(Method changeSetMethod, Object changeLogInstance) throws BeansException, IllegalAccessException, InvocationTargetException {
```

Comment:
I understand your intention(at least one of them) is to be able to inject the required dependencies(DB and mongoTemplate) via Spring context.

However with this approach you are not restricted to only inject DB and MongoTemplate, you could inject anything. Is this your goal too?

Reply:
The ability to inject the original parameters was just a happy side effect of our goal. The real goal was to remove the restriction on the parameters you could pass altogether. In our use case, for example, we need to access data from another service and need to inject an ApiServiceClientFacade (internal architecture). The object we need requires other data on the context so we could not just construct one at the time a change log was created.

Reply:
Cool, that's make more sense as the idea of doing all this just to pass DB and MongoTemplate in the context doesn't really add much value.